packages/components/src/components/post-tab-panel/post-tab-panel.tsx

```diff
@@ -0,0 +1,78 @@
+import { Component, Element, h, Method, State, Watch } from '@stencil/core';
+import { checkExists, getChildIndex, onTransitionEnd } from '../../utils';
+
+@Component({
+  tag: 'post-tab-panel',
+  styleUrl: 'post-tab-panel.scss',
+  shadow: true,
+})
+export class PostTabPanel {
+  private tabPanelElement: HTMLElement;
+
+  @Element() host: HTMLPostTabPanelElement;
+
+  @State() isShown = true;
+  @State() tabIndex: number;
+
+  @State() associatedHeader: HTMLPostTabHeaderElement;
+
+  @Watch('associatedHeader')
+  initPanelVisibility(associatedHeader: HTMLPostTabHeaderElement) {
+    if (!associatedHeader.active) {
+      this.isShown = false;
+      this.host.remove();
+    }
+  }
+
+  @State() tabGroup: HTMLPostTabsElement;
+
+  @Watch('tabGroup')
+  validateTabGroup(newValue: HTMLPostTabsElement | undefined) {
+    checkExists(newValue, 'The post-tab-panel should be contained in a post-tabs element.');
+  }
+
+  componentWillLoad() {
+    this.tabIndex = getChildIndex(this.host, 'post-tab-panel');
+    this.tabGroup = this.host.closest('post-tabs');
+    this.associatedHeader = this.host.previousElementSibling as HTMLPostTabHeaderElement;
+  }
+
+  componentDidLoad() {
+    this.tabPanelElement = this.host.shadowRoot.querySelector('.tab-pane');
+  }
+
+  /**
+   * Shows the tab panel programmatically.
+   */
+  @Method()
+  async show() {
+    this.associatedHeader.insertAdjacentElement('afterend', this.host);
+    setTimeout(() => {
+      this.isShown = true;
+    });
+  }
+
+  /**
+   * Hides the tab panel programmatically.
+   */
+  @Method()
+  async hide() {
+    this.isShown = false;
+    await onTransitionEnd(this.tabPanelElement).then(() => {
+      this.host.remove();
+    });
```

Comment:
Will `transitionend` be fired if the user prefers reduced motion? We could use the [web animation API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API/Using_the_Web_Animations_API) to do the transitions. This will return a promise that resolves anyways.

Reply:
I wrote the [onTransitionEnd](https://github.com/swisspost/design-system/blob/main/packages/components/src/utils/on-transition-end.ts) function to return a promise that directly resolves if the user prefers reduced motion but waits for the transition to end otherwise.

The Web Animations API is also a good alternative but it involves rewriting Bootstrap animations.

Reply:
I see. There is one more benefit I used in the header. If the animation should be cancelled because the user cancelled the action (clicking everywhere like a madman), the animation can be reversed instantly by calling `animation.reverse()`. This could lead to snappier interactions because orchestration is easier and predictable. We could discuss this next Thursday.